### ISSUE TITLE
6>7 Pokemon now have the flag "Invalid: HT Memory is missing."

### ISSUE THREAD

**Issue by Kukuikukuikukui** [kwsch/PKHeX#783]

**Comment by ReignOfComputer** [comment]
Which build are you using?

**Comment by CoSMiiCKiiD** [comment]
![image](https://cloud.githubusercontent.com/assets/25294977/22456716/175390cc-e7f9-11e6-8ce1-593f59c82c57.png)

![image](https://cloud.githubusercontent.com/assets/25294977/22456739/36c8f2e4-e7f9-11e6-8132-496f4704044f.png)

Latest build im getting this issue, I cant compile my own PKHEx so i download the latest update versions any help would be greatly appreciated!

**Comment by ReignOfComputer** [comment]
Try updating your PKHeX version and see if that fixes things. You're not on the latest build.

https://projectpokemon.org/forums/files/file/1-pkhex/

**Comment by CoSMiiCKiiD** [comment]
thanks reign

**Comment by ReignOfComputer** [comment]
Did it work?

**Comment by CoSMiiCKiiD** [comment]
Exception Details:
System.MissingMethodException: Method not found: '!!0[] System.Array.Empty()'.
   at PKHeX.Core.LegalityAnalysis.getLegalityReport()
   at PKHeX.WinForms.Main.showLegality(PKM pk, Boolean tabs, Boolean verbose, Boolean skipMoveRepop)
   at PKHeX.WinForms.Main.clickLegality(Object sender, EventArgs e)
   at System.Windows.Forms.ToolStripItem.RaiseEvent(Object key, EventArgs e)
   at System.Windows.Forms.ToolStripMenuItem.OnClick(EventArgs e)
   at System.Windows.Forms.ToolStripItem.HandleClick(EventArgs e)
   at System.Windows.Forms.ToolStripItem.HandleMouseUp(MouseEventArgs e)
   at System.Windows.Forms.ToolStripItem.FireEventInteractive(EventArgs e, ToolStripItemEventType met)
   at System.Windows.Forms.ToolStripItem.FireEvent(EventArgs e, ToolStripItemEventType met)
   at System.Windows.Forms.ToolStrip.OnMouseUp(MouseEventArgs mea)
   at System.Windows.Forms.ToolStripDropDown.OnMouseUp(MouseEventArgs mea)
   at System.Windows.Forms.Control.WmMouseUp(Message& m, MouseButtons button, Int32 clicks)
   at System.Windows.Forms.Control.WndProc(Message& m)
   at System.Windows.Forms.ScrollableControl.WndProc(Message& m)
   at System.Windows.Forms.ToolStrip.WndProc(Message& m)
   at System.Windows.Forms.ToolStripDropDown.WndProc(Message& m)
   at System.Windows.Forms.Control.ControlNativeWindow.OnMessage(Message& m)
   at System.Windows.Forms.Control.ControlNativeWindow.WndProc(Message& m)
   at System.Windows.Forms.NativeWindow.Callback(IntPtr hWnd, Int32 msg, IntPtr wparam, IntPtr lparam)

Loaded Assemblies:
--------------------
mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089
C:\Windows\Microsoft.NET\Framework\v4.0.30319\mscorlib.dll

PKHeX-mono, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null
C:\Users\Master\AppData\Local\Temp\Temp1_PKHeX-mono (170131).zip\PKHeX.exe

System, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089
C:\Windows\Microsoft.Net\assembly\GAC_MSIL\System\v4.0_4.0.0.0__b77a5c561934e089\System.dll

System.Windows.Forms, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089
C:\Windows\Microsoft.Net\assembly\GAC_MSIL\System.Windows.Forms\v4.0_4.0.0.0__b77a5c561934e089\System.Windows.Forms.dll

System.Drawing, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a
C:\Windows\Microsoft.Net\assembly\GAC_MSIL\System.Drawing\v4.0_4.0.0.0__b03f5f7f11d50a3a\System.Drawing.dll

System.Core, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089
C:\Windows\Microsoft.Net\assembly\GAC_MSIL\System.Core\v4.0_4.0.0.0__b77a5c561934e089\System.Core.dll

System.Configuration, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a
C:\Windows\Microsoft.Net\assembly\GAC_MSIL\System.Configuration\v4.0_4.0.0.0__b03f5f7f11d50a3a\System.Configuration.dll

System.Xml, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089
C:\Windows\Microsoft.Net\assembly\GAC_MSIL\System.Xml\v4.0_4.0.0.0__b77a5c561934e089\System.Xml.dll

--------------------
User Message:
An unhandled exception has occurred.
You can continue running PKHeX, but please report this error.

**Comment by CoSMiiCKiiD** [comment]
literally every single pokemon gets this now on new updated version

**Comment by CoSMiiCKiiD** [comment]
![image](https://cloud.githubusercontent.com/assets/25294977/22457870/c2ce72c8-e7fe-11e6-8778-0220c247df4d.png)

downloaded and installed again same thing, mew is a pk file from orginal gen, but now nearly all non gen 7 pokemon are getting, also my legitmate trevant is getting flagged for some super train flag

**Comment by CoSMiiCKiiD** [comment]
sorry all pokemon are getting

**Comment by ReignOfComputer** [comment]
Oh wow okay.

Are you sure you've overwritten the PKHeX.Core.dll with the new one?

**Comment by CoSMiiCKiiD** [comment]
There wasn't a new core.dll in my new zip folder?, should there been?

![image](https://cloud.githubusercontent.com/assets/25294977/22458794/ecaacc46-e802-11e6-93ce-4c4621fad25b.png)

**Comment by ReignOfComputer** [comment]
My bad, yeah there isn't one bundled with the new release.

No idea what's going on here, sorry.

**Comment by ReignOfComputer** [comment]
Oh wait I see what's going on. Your computer doesn't have a compatible version of .NET Framework.

You're running Windows 7, right?

**Comment by CoSMiiCKiiD** [comment]
oh yup i am windows 7, is the net framework addition just been updated? as the previous PKHEX worked fine,?

**Comment by CoSMiiCKiiD** [comment]
updated and the crash seems to go now but the same thing above is happening with the HT Memory, would a new core dll fix this? and if so anyone got a file link i can have?

**Comment by egzonqj** [comment]
Seems like both .exe and .dll are merged in the same .exe in this release.

**Comment by CoSMiiCKiiD** [comment]
OK thanks Egzonqj for the reply and thanks Reign for the patience !
Besides the Mew HT Memory not working on this build my Legitimate ungenned treveant is getting this error also.

![image](https://cloud.githubusercontent.com/assets/25294977/22462847/dcafdc6c-e813-11e6-89d1-6d40854d45d7.png)

any suggestions whats causing the two?

**Comment by CoSMiiCKiiD** [comment]
I havent Super trained this guy literally just adjusted level etc

**Comment by egzonqj** [comment]
I think there was a button added in the recent releases to remove the medals of supertraining, which are not used in gen7, iirc?

**Comment by ReignOfComputer** [comment]
Try [OT/Misc] > [Medals] > [Remove All].

**Comment by kwsch** [comment]
I'm a little suspicious that this offset is used by something new; transfer method clears them but who knows if gen7 sets something (ie with Pelago).

**Comment by egzonqj** [comment]
Is there a way to check that, isn't it?

**Comment by CoSMiiCKiiD** [comment]
Once again Reign thanks fore trevenant it worked! Im still gettng this HT memory thing with mew though, Im using slayers Pk file from project pokemon. Any ideas?

![image](https://cloud.githubusercontent.com/assets/25294977/22481319/adf54e48-e859-11e6-888c-cd9253c0c29a.png)

**Comment by sora10pls** [comment]
Values should be equal to these.

![image](https://cloud.githubusercontent.com/assets/17801814/22497240/2bd57cf2-e81e-11e6-8c59-3fce3eab1895.png)

**Comment by Kukuikukuikukui** [comment]
Every single Pokemon that I transferred from XY/ORAS using Pokemon Bank has this flag. The transferred are almost 100% legit, I caught a random Pokemon in the wild with a standard PokeBall and transferred it to SM, to test if this is really an error or just an error with the Pokemon I am using, and this Pokemon is deemed illegal by PKHeX. Note that I am checking the legality using the (CTRL + RIGHT CLICK into legality) option, since if you select the Pokemon and do the legality check in the PK file editor, it will automatically fix the illegal Pokemon and make it legal, not displaying that flag.

**Comment by CoSMiiCKiiD** [comment]
sora10pls thanks so much that worked and i can finally get back to my game!

**Comment by SonicBlader** [comment]
I had the same issue and I managed to fix it. I was trying to change the level of the VC Mew .pk7 file which I had downloaded from ProjectPokemon Event Gallery. The pokemon has no details in the NotOT field. When I tried changing the level I got the same error. Even after providing the NotOT details the legality checker persisted. I reloaded the .pk7 file and enter the NotOT details with my OT. I was then able to edit the levels and other details without any problems.
